src/main/java/com/boomi/flow/services/boomi/mdh/database/FieldMapper.java

```diff
@@ -109,34 +108,84 @@
+        Universe.Layout.Model.Element foundElement = elements.stream()
+                .filter(element -> element.getName().equals(property.getDeveloperName()))
+                . findFirst()
+                .orElseGet(Universe.Layout.Model.Element::new);
+
+        if (foundElement.isRepeatable()) {
+            return foundElement.getCollectionUniqueId().toLowerCase();
+        } else {
+            return property.getDeveloperName();
+        }
+    }
+
+    static Object createMapEntry(Property property, List<Universe.Layout.Model.Element> elements) {
+        if (property.getContentValue() != null) {
             if (property.getContentType() == ContentType.DateTime && Strings.isNullOrEmpty(property.getContentValue())) {
                 // Ignore datetime with empty values
-                continue;
+                return null;
             }
 
-            if (property.getContentValue() != null) {
-                if (property.getContentType() == ContentType.DateTime) {
-                    String dateFormatted = OffsetDateTime
-                            .parse(property.getContentValue())
-                            .format(DateTimeFormatter.ISO_DATE_TIME.withZone(ZoneId.of("Z")));
+            if (property.getContentType() == ContentType.DateTime) {
+                return OffsetDateTime
+                        .parse(property.getContentValue())
+                        .format(DateTimeFormatter.ISO_DATE_TIME.withZone(ZoneId.of("Z")));
+            } else {
+                return property.getContentValue();
+            }
+        } else if (property.getObjectData() != null) {
+            List<Map<String, Object>> listOfObjects = new ArrayList<>();
 
-                    mapObject.put(property.getDeveloperName(), dateFormatted);
-                } else {
-                    mapObject.put(property.getDeveloperName(), property.getContentValue());
-                }
-            } else if (property.getObjectData() != null && property.getObjectData().size() > 0) {
-                mapObject.put(property.getDeveloperName(), createMapFromMobject(property.getObjectData().get(0)));
+            for (MObject mobjectItem: property.getObjectData()) {
+                listOfObjects.add(createMapFromMobject(mobjectItem, elements));
             }
+
+            return listOfObjects;
         }
+        return null;
+    }
 
-        return mapObject;
+    public static Map<String, Object> createMapFromMobject(MObject mObject, List<Universe.Layout.Model.Element> elements) {
+        Map<String, Object> mapObject = new HashMap<>();
+
+        for (Property property: mObject.getProperties()) {
+            mapObject.put(property.getDeveloperName(), createMapEntry(property, elements));
+        }
+
+        Map<String, Object> wrapperObject = new HashMap<>();
+        wrapperObject.put(mObject.getDeveloperName().substring(0, mObject.getDeveloperName().length()-6), mapObject);
```

Comment:
What are we stripping here?

Reply:
we are stripping "-child" at the end of child objects.  The name of the child object has been already modified in a new pull request to be like "model - fieldGroupRepeateble"